<a href="https://colab.research.google.com/github/ocintho/IGS/blob/main/exercice_pandas.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [44]:
import pandas as pd


Lire les données du fichier CSV ‘dataset-sell4all.csv’ qui contient les données
démographiques et les dépenses des utilisateurs,


In [45]:
#Lire les données du fichier CSV ‘dataset-sell4all.csv’ qui contient les données démographiques et les dépenses des utilisateurs,
dataset=pd.read_csv("dataset-sell4all-maroc.csv")

In [46]:
#Afficher des informations sur les 5 premières lignes du fichier CSV
dataset.head(5)

Name    Phone Number                            Email  \
0       Aaron Cote        966-7625                 elit@hotmail.org   
1  Angelica Lawson        232-3051            diam.proin@google.org   
2    Louis Gilbert  1-997-733-0134      lorem.fringilla@hotmail.org   
3     Basia Finley  1-987-322-7148      tristique.aliquet@icloud.co   
4      Rhona Sears        387-7682  iaculis.odio.nam@protonmail.org   

                  Address       Country Postal code Last date of connection  \
0   699-5837 Risus Street        Norway        4126             Apr 5, 2021   
1  481-8428 Magna. Street      Pakistan      518885            Oct 10, 2021   
2  Ap #192-2082 Enim. Ave      Colombia      575444             Jul 2, 2022   
3        608-2732 Nec Rd.  South Africa   5973-5765            Feb 19, 2022   
4             8763 In Rd.        France       37476            Mar 31, 2022   

  Last time of connection  Age Gender  Customer spendings  
0                 4:39 AM   71    Man                 356  
1                12:36 AM   37  Women                 173  
2                 3:37 AM   24  Women                 105  
3                 2:34 AM   37  Women                  28  
4                 6:28 AM   42  Women                  13

In [47]:
#nombre de lignes et colones

dataset.values.shape


(505, 11)

In [48]:
#les colones du fichier CSV
dataset.columns

Index(['Name', 'Phone Number', 'Email', 'Address', 'Country', 'Postal code',
       'Last date of connection', 'Last time of connection', 'Age', 'Gender',
       'Customer spendings'],
      dtype='object')

In [49]:
#les types de données des champs du fichier CSV
dataset.dtypes


Name                       object
Phone Number               object
Email                      object
Address                    object
Country                    object
Postal code                object
Last date of connection    object
Last time of connection    object
Age                         int64
Gender                     object
Customer spendings          int64
dtype: object

In [ ]:
~#Expliquer les détails affichés du résumé technique dans une cellule de démarque du
#bloc-notes Jupyter,

d'après les informations precedente nous voyons que dans notre etude se porte sur 505 clients avec 11 critères different qui sont: Name, Phone Number, Email, Address, Country, Postal code,Last date of connection, Last time of connection, Age, Gender,Customer spendings et a part l'age et les depences consomateur le reste des donné sont de valeur qualitatives

In [51]:
#Calculer la médiane et la moyenne des colonnes
print("mediane age",dataset["Age"].median())
print("mediane Customer spendings",dataset["Customer spendings"].median())

print("moyenne age",dataset["Age"].mean())
print("moyenne Customer spendings",dataset["Customer spendings"].mean())



mediane age 46.0
mediane Customer spendings 307.0
moyenne age 46.08316831683168
moyenne Customer spendings 311.16831683168317


In [40]:
#recoder l’age en variable qualitative ne faisant des inter

def categories_age(age):
    if age < 18:
        return 'mineur'
    elif age < 38:
        return 'jeune'
    elif age < 68:
        return 'adulte'
    else:
        return 'vieux'

dataset['categorie age'] = dataset['Age'].map(categories_age)
dataset.drop(['Age'], axis=1, inplace=True)

In [41]:
#Nettoyer les lignes avec moins de 10 € de dépenses par client : supprimez toutes les
#lignes d’utilisateurs ayant dépensé moins de 10 € sur le site,

dataset.drop( dataset[ (dataset['Customer spendings'] < 10)].index, inplace=True)
dataset

Name    Phone Number                            Email  \
0          Aaron Cote        966-7625                 elit@hotmail.org   
1     Angelica Lawson        232-3051            diam.proin@google.org   
2       Louis Gilbert  1-997-733-0134      lorem.fringilla@hotmail.org   
3        Basia Finley  1-987-322-7148      tristique.aliquet@icloud.co   
4         Rhona Sears        387-7682  iaculis.odio.nam@protonmail.org   
..                ...             ...                              ...   
500    Abraham Walker        835-6185        nunc.sed.orci@outlook.net   
501      Drake Carney        891-0713    dignissim.pharetra@icloud.org   
502      Hadley Sykes  1-865-267-6254                luctus@icloud.com   
503      Dean Hensley  1-776-366-7784                      dui@aol.com   
504  Margaret Leblanc        522-8552               nec.enim@yahoo.org   

                           Address        Country Postal code  \
0            699-5837 Risus Street         Norway        4126   
1           481-8428 Magna. Street       Pakistan      518885   
2           Ap #192-2082 Enim. Ave       Colombia      575444   
3                 608-2732 Nec Rd.   South Africa   5973-5765   
4                      8763 In Rd.         France       37476   
..                             ...            ...         ...   
500            Ap #258-8540 A, St.   South Africa   3181-4474   
501             304-4217 Ac Street         France       54615   
502    P.O. Box 851, 819 Pede. Av.           Peru       16373   
503  P.O. Box 896, 2956 Risus. Ave          China      746013   
504            437-1106 Nec Street  United States       74253   

    Last date of connection Last time of connection Gender  \
0               Apr 5, 2021                 4:39 AM    Man   
1              Oct 10, 2021                12:36 AM  Women   
2               Jul 2, 2022                 3:37 AM  Women   
3              Feb 19, 2022                 2:34 AM  Women   
4              Mar 31, 2022                 6:28 AM  Women   
..                      ...                     ...    ...   
500            Jan 31, 2023                 1:46 PM  Women   
501            Jun 11, 2022                10:06 AM  Women   
502            Jun 13, 2022                 6:49 PM  Women   
503            Aug 27, 2021                 6:02 AM  Women   
504             Feb 2, 2023                 4:56 AM  Women   

     Customer spendings categorie age  
0                   356         vieux  
1                   173         jeune  
2                   105         jeune  
3                    28         jeune  
4                    13        adulte  
..                  ...           ...  
500                 182         jeune  
501                  25        adulte  
502                  42        adulte  
503                 511         jeune  
504                 518         vieux  

[497 rows x 11 columns]

In [42]:
#Nettoyer les doublons : supprimer toutes les lignes qui apparaissent plus d’une fois
#dans les données,
dataset.drop_duplicates( keep = 'first' , inplace = True , ignore_index = False )
